public-ui/src/app/views/pm/pm.component.ts

```diff
@@ -1,15 +1,228 @@
+
+  loading() {
+    return this.loadingStack.length > 0;
+  }
 
   ngOnInit() {
+    this.showMeasurement(this.selectedGroup, this.selectedConcept);
+    // Get demographic totals
+    this.loadingStack.push(true);
+    this.subscriptions.push(this.api.getGenderAnalysis()
+      .subscribe(result => {
+        this.genderAnalysis = result;
+        for (const g of this.genderAnalysis.results) {
+          if (g.stratum1 === this.dbc.FEMALE_GENDER_ID) {
+            this.femaleCount = g.countValue;
+          } else if (g.stratum1 === this.dbc.MALE_GENDER_ID) {
+            this.maleCount = g.countValue;
+          } else {
+            this.otherCount += g.countValue;
+          }
+        }
+        this.loadingStack.pop();
+      }));
+
+    this.loadingStack.push(true);
+    this.subscriptions.push(this.api.getRaceAnalysis()
+      .subscribe(result => {
+        this.raceAnalysis = result;
+        this.loadingStack.pop();
+      }));
+
+    this.loadingStack.push(true);
+    this.subscriptions.push(this.api.getEthnicityAnalysis()
+      .subscribe(result => {
+        this.ethnicityAnalysis = result;
+        this.loadingStack.pop();
+      }));
+  }
+
+  ngOnDestroy() {
+    for (const s of this.subscriptions) {
+      s.unsubscribe();
+    }
+  }
+
+  showMeasurement(group: any, concept: any) {
+    this.selectedGroup = group;
+    this.selectedConcept = concept;
+
+    if (!this.selectedConcept.analyses) {
+      this.loadingStack.push(true);
+      this.subscriptions.push(this.api.getConceptAnalysisResults(
+        [this.selectedConcept.conceptId])
+        .subscribe(result => {
+          this.selectedConcept.analyses = result.items[0];
+          // Organize, massage the data for ui graphing
+          // for example, pregnant has only 1 result for pregnant,
+          // we add a not pregnant to make display better
+          this.arrangeConceptAnalyses(this.selectedConcept);
+          this.loadingStack.pop();
+      }));
+    } else {
+      // Don't get the analyses
+    }
+  }
+
+  arrangeConceptAnalyses(concept: any) {
+
+    if (concept.conceptId === this.dbc.PREGNANCY_CONCEPT_ID) {
+      // Delete any male results so we don't look dumb with dumb data
+      concept.analyses.measurementValueMaleAnalysis = null;
+      concept.analyses.measurementValueOtherGenderAnalysis = null;
+      concept.analyses.genderAnalysis.results =
+        concept.analyses.genderAnalysis.results.filter(result =>
+        result.stratum2 === this.dbc.FEMALE_GENDER_ID );
+
+      // Add not pregnant result to the female value results
+      // because this concept is just a one value Yes
+      const pregnantResult  = concept.analyses.measurementValueFemaleAnalysis.results[0];
+
+      const notPregnantResult: AchillesResult = {
+        analysisId: pregnantResult.analysisId,
+        stratum1: pregnantResult.stratum1,
+        stratum2: pregnantResult.stratum2,
+        stratum3: pregnantResult.stratum3,
+        stratum4: 'Not Pregnant',
```

Comment:
WHy is this one hardcoded? Comment?

Reply:
I could use a constant. We don't get any result for not pregnant. It is an analysis with one result with the count of people that are pregnant. But that doesn't make for a good graph. So i make a not pregnant result here. This needs a little more design about how to graph concepts like "pregnant" and "wheel chair use" that just give us the count for those things.  I just threw this in there for a option for now.